XenModel/Actions/OvfActions/ImportImageAction.cs

```diff
@@ -43,33 +43,26 @@
 
 namespace XenAdmin.Actions.OvfActions
 {
-	public class ImportImageAction : ApplianceAction
+	public class ImportImageAction : ImportApplianceAction
 	{
 		#region Private fields
 
 		private readonly EnvelopeType m_ovfEnvelope;
-		private readonly Dictionary<string, VmMapping> m_vmMappings;
-		private readonly bool m_runfixups;
-		private readonly SR m_selectedIsoSr;
-		private readonly string m_directory;
+        private readonly string m_directory;
 
 		#endregion
 
 		public ImportImageAction(IXenConnection connection, EnvelopeType ovfEnv, string directory, Dictionary<string, VmMapping> vmMappings, bool runfixups, SR selectedIsoSr)
-			: base(connection, string.Format(Messages.IMPORT_DISK_IMAGE, ovfEnv.Name, Helpers.GetName(connection)))
+			: base(connection, null, vmMappings, false, false, null, runfixups, selectedIsoSr)
 		{
 			m_ovfEnvelope = ovfEnv;
 			m_directory = directory;
-			m_vmMappings = vmMappings;
-			m_runfixups = runfixups;
-			m_selectedIsoSr = selectedIsoSr;
-		}
-
-		protected override void Run()
-		{
-		    base.Run();
+            Title = string.Format(Messages.IMPORT_DISK_IMAGE, ovfEnv.Name, Helpers.GetName(connection));
+        }
 
-			Debug.Assert(m_vmMappings.Count == 1, "There is one VM mapping");
+        protected override void RunCore()
+        {
+            Debug.Assert(m_vmMappings.Count == 1, "There is one VM mapping");

```

Comment:
Nit: The assertion failure message ought to read `"There must be only one VM mapping"`.
This was the case before the changes here however.

Reply:
Will correct it in the upcoming PR.